mmcv/runner/hooks/evaluation.py

```diff
@@ -150,7 +153,7 @@ def __init__(self,
         self.out_dir = out_dir
         self.file_client_args = file_client_args
 
-    def _init_rule(self, rule, key_indicator):
+    def _init_rule(self, rule: Optional[str], key_indicator: Optional[str]):
```

Comment:
It seems there is not default `None` value for  `rule` and `key_indicator`. 
```suggestion
    def _init_rule(self, rule: Optional[str] = None, key_indicator: str):
```
Will this be more properer?

Reply:
I don't think it will work since positional arguments should come first before keyword arguments   
I actually just followed the function docs here, but from the use case I would think it would be better to remove both the `Optional` type hints. what do you think?

Reply:
Agree. I think the docstring is correct. However it forgets to give the default value for `rule`. BTW, could it passes through mypy if we simply setting rule as `rule: str`?

Reply:
apparently it does not passes, so I think the `rule` still needs the optional but `key_indicator` is not. How do you think?